### PR TITLE
Don't crash the entire site when a singular result is null

### PR DESCRIPTION
**Pull request by ghost**
# Why

When searching for "alan", a singular result would come back null.

# What

This PR fixes the above by doing a flatmap, mapping to a result component if the result is not null, and mapping to nothing if the result is null.

### CODE REVIEW

File: frontend/components/ResultsPage/ResultsLoader.tsx [hunk 1]
```diff
@@ -24,12 +24,12 @@ function ResultsLoader({ results, loadMore }: ResultsLoaderProps) {
     >
       <div className='five column row'>
         <div className='page-home'>
-          {results.map((result) => (
+          {results.flatMap((result) => (result ? (
```
Comment (gamburgm) [inline frontend/components/ResultsPage/ResultsLoader.tsx]: Should probably do a `filter` to filter out nulls then a `map` to map the remaining results to the React jazz

**Review by gamburgm — CHANGES_REQUESTED** [review]
At a high-level, I think we should figure out why a `null` is being returned from the backend first before merging this, b/c that might be due to a search result not getting properly fetched from the DB, or some other breaking issue there.

File: frontend/components/ResultsPage/ResultsLoader.tsx (hunk 1 above)
Comment (ghost) [reply frontend/components/ResultsPage/ResultsLoader.tsx]: done

**Review by NEUDitao — APPROVED** [review]
I think either way we *should* push out a fix now, since the website is crashing whenever anyone searches "Alan," and at the very least put a TODO linking to Mitch's comment in the code. Yes -- it'd be nice to figure out why null results are happening, but we don't have that type of time.

**Pull request merged**